packages/consensus/finality/finality_gadget_test.go

```diff
@@ -0,0 +1,393 @@
+					gof.Low:  {"Message7"},
+					gof.None: {"Message6"},
+				})
+			},
+		},
+	}
+	for i := 0; processMsgScenario.HasNext(); i++ {
+		if len(prePostSteps)-1 < i {
+			processMsgScenario.Next(nil)
+			continue
+		}
+		processMsgScenario.Next(prePostSteps[i])
+	}
+}
+
+func assertMsgsGoFs(t *testing.T, testFramework *tangle.MessageTestFramework, expected map[gof.GradeOfFinality][]string) {
+	for expectedGoF, msgAliases := range expected {
+		for _, msgAlias := range msgAliases {
+			actualGradeOfFinality := testFramework.MessageMetadata(msgAlias).GradeOfFinality()
+			assert.Equal(t, expectedGoF, actualGradeOfFinality, "expected msg %s GoF to be %s but is %s", msgAlias, expectedGoF, actualGradeOfFinality)
+		}
+	}
+}
+
+func assertTxsGoFs(t *testing.T, testFramework *tangle.MessageTestFramework, expected map[gof.GradeOfFinality][]string) {
+	for expectedGoF, msgAliases := range expected {
+		for _, msgAlias := range msgAliases {
+			txMeta := testFramework.TransactionMetadata(msgAlias)
+			actualGradeOfFinality := txMeta.GradeOfFinality()
+			assert.Equal(t, expectedGoF, actualGradeOfFinality, "expected tx %s (via msg %s) GoF to be %s but is %s", txMeta.ID(), msgAlias, expectedGoF, actualGradeOfFinality)
+			// auto. also check outputs
+			for _, output := range testFramework.Transaction(msgAlias).Essence().Outputs() {
+				outputGoF := testFramework.OutputMetadata(output.ID()).GradeOfFinality()
+				assert.Equal(t, expectedGoF, outputGoF, "expected also tx output %s (via msg %s) GoF to be %s but is %s", output.ID(), msgAlias, expectedGoF, outputGoF)
+			}
+		}
+	}
+}
+
+func assertBranchsGoFs(t *testing.T, testFramework *tangle.MessageTestFramework, expected map[gof.GradeOfFinality][]string) {
+	for expectedGoF, msgAliases := range expected {
+		for _, msgAlias := range msgAliases {
+			branch := testFramework.Branch(msgAlias)
+			actualGradeOfFinality := branch.GradeOfFinality()
+			assert.Equal(t, expectedGoF, actualGradeOfFinality, "expected branch %s (via msg %s) GoF to be %s but is %s", branch.ID(), msgAlias, expectedGoF, actualGradeOfFinality)
+		}
+	}
+}
+
+func wireUpEvents(t *testing.T, testTangle *tangle.Tangle, fg Gadget) {
+	testTangle.ApprovalWeightManager.Events.MarkerWeightChanged.Attach(events.NewClosure(func(e *tangle.MarkerWeightChangedEvent) {
+		if err := fg.HandleMarker(e.Marker, e.Weight); err != nil {
+			t.Log(err)
+		}
+	}))
+	testTangle.ApprovalWeightManager.Events.BranchWeightChanged.Attach(events.NewClosure(func(e *tangle.BranchWeightChangedEvent) {
+		if err := fg.HandleBranch(e.BranchID, e.Weight); err != nil {
+			t.Log(err)
+		}
+	}))
+
+	// we need to update the WeightProvider on confirmation
+	fg.Events().MessageConfirmed.Attach(events.NewClosure(func(messageID tangle.MessageID) {
+		testTangle.Storage.Message(messageID).Consume(func(message *tangle.Message) {
+			testTangle.WeightProvider.Update(message.IssuingTime(), identity.NewID(message.IssuerPublicKey()))
+		})
+	}))
```

Comment:
Do we really need to do this? Doesn't it just use the static weight defined in the scenario?

Reply:
Removed as no longer necessary